modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/types/BLangFunctionTypeNode.java

```diff
@@ -68,13 +68,34 @@ public String toString() {
         StringBuilder br = new StringBuilder();
         br.append("function(");
         if (paramTypeNodes.size() > 0) {
-            br.append(Arrays.toString(paramTypeNodes.toArray()));
+            br.append(getParamNames(paramTypeNodes));
         }
         if (returnParamTypeNodes.size() > 0) {
             br.append(returnsKeywordExists ? ")returns(" : ")(");
-            br.append(Arrays.toString(returnParamTypeNodes.toArray()));
+            br.append(getParamNames(returnParamTypeNodes));
         }
         br.append(")");
         return br.toString();
     }
+
+    private String getParamNames(List<TypeNode> paramTypes) {
+        StringBuilder sb = new StringBuilder();
+        int nParamTypes = paramTypes.size();
+
+        for (int i = 0; i < nParamTypes; i++) {
+            TypeNode paramType = paramTypes.get(i);
+            if (paramType.getKind() == NodeKind.USER_DEFINED_TYPE) {
+                sb.append(((UserDefinedTypeNode) paramType).getTypeName().getValue());
+            } else {
+                sb.append(paramType.toString());
+            }
+
+            if (i == nParamTypes - 1) {
```

Comment:
we can use [StringJoiner](https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html) instead of StringBuilder, to do this in a much simpler way.